src/render/PanelHeaderProvider.js

```diff
@@ -66,58 +66,55 @@ export function getConcreteType(element) {
   return type;
 }
 
-export const PanelHeaderProvider = {
+export const PanelHeaderProvider = (iconProperty) => {
+  return {
+    getDocumentationRef: (element) => {
+      const elementTemplates = getTemplatesService();
 
-  getDocumentationRef: (element) => {
-    const elementTemplates = getTemplatesService();
-
-    if (elementTemplates) {
-      return getTemplateDocumentation(element, elementTemplates);
-    }
-  },
-
-  getElementLabel: (element) => {
-    if (is(element, 'bpmn:Process')) {
-      return getBusinessObject(element).name;
-    }
-
-    return getLabel(element);
-  },
+      if (elementTemplates) {
+        return getTemplateDocumentation(element, elementTemplates);
+      }
+    },
 
-  getElementIcon: (element) => {
-    const concreteType = getConcreteType(element);
+    getElementLabel: (element) => {
+      if (is(element, 'bpmn:Process')) {
+        return getBusinessObject(element).name;
+      }
 
-    const elementTemplates = getTemplatesService();
+      return getLabel(element);
+    },
 
-    if (elementTemplates) {
-      const template = getTemplate(element, elementTemplates);
+    getElementIcon: (element) => {
+      const concreteType = getConcreteType(element);
 
-      if (template && template.icon) {
-        return () => <img class="bio-properties-panel-header-template-icon" width="32" height="32" src={ template.icon.contents } />;
+      const templateIcon = getModelerTemplateIcon(element, iconProperty);
```

Comment:
Instead of passing `iconProperty` all the way down you can use the `useService` (which was also used previously) hook to get hold of it:

```
const config = useService('config.elementTemplateIconRenderer', false);

if (config) {
  ...
}
```

Reply:
@philippfromme Thank you very much for the reminder.It looks perfect now😄

Reply:
Much better. 😉